unittests/etl/CacheLoaderTest.cpp

```diff
@@ -0,0 +1,141 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <etl/impl/CacheLoader.h>
+#include <util/Fixtures.h>
+#include <util/MockCache.h>
+
+#include <boost/asio.hpp>
+#include <boost/json.hpp>
+#include <gtest/gtest.h>
+
+namespace json = boost::json;
+using namespace clio::detail;
+using namespace clio;
+using namespace Backend;
+using namespace testing;
+
+constexpr static auto SEQ = 30;
+constexpr static auto INDEX1 = "E6DBAFC99223B42257915A63DFC6B0C032D4070F9A574B255AD97466726FC321";
+
+struct CacheLoaderTest : public MockBackendTest
+{
+    void
+    SetUp() override
+    {
+        MockBackendTest::SetUp();
+        work.emplace(ctx);
+        for (auto i = 0; i < 2; ++i)
+            optThreads.emplace_back([&] { ctx.run(); });
+    }
+
+    void
+    TearDown() override
+    {
+        work.reset();
+        for (auto& optThread : optThreads)
+            if (optThread.joinable())
+                optThread.join();
+        ctx.stop();
+        MockBackendTest::TearDown();
+    }
+
+protected:
+    MockCache cache;
+    Config cfg{json::parse("{}")};
+    std::optional<boost::asio::io_service::work> work;
+    boost::asio::io_context ctx;
+    std::vector<std::thread> optThreads;
+};
+
+namespace {
+
+std::vector<LedgerObject>
+getLatestDiff()
+{
+    return std::vector<LedgerObject>{
+        {.key = ripple::uint256{"05E1EAC2574BE082B00B16F907CE32E6058DEB8F9E81CF34A00E80A5D71FA4FE"}, .blob = Blob{'s'}},
```

Comment:
I did not know that designated initializers are already supported on AppleClang 🥳 
We should check if they are also working on gcc11 and regular clang that we advertise as supported before we can merge this. I vividly remember breaking the build with these before we had CI.. and as you know right now our CI is Mac only so we gotta check manually..

Reply:
clang 10 (latest is 15) ,Apple clang 12 (latest is 14), Gcc 8 (latest is 13) should support.
I checked on godbolt. should be okay